reactor-netty-http/src/test/java/reactor/netty/http/HttpProtocolsTests.java

```diff
@@ -551,4 +555,42 @@ private void doTestTrailerHeaders(HttpClient client, String expectedHeaderValue,
 		      .expectComplete()
 		      .verify(Duration.ofSeconds(5));
 	}
+
+	@ParameterizedCompatibleCombinationsTest
+	void testIdleTimeoutAddedCorrectly(HttpServer server, HttpClient client) {
+		ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+		Logger logger = (Logger) LoggerFactory.getLogger(DefaultPromise.class);
+		logger.addAppender(listAppender);
+		listAppender.start();
+
+		disposableServer =
+				server.idleTimeout(Duration.ofSeconds(60))
+						.route(routes ->
+								routes.post("/echo", (req, res) -> res.send(req.receive().retain())))
+						.bindNow();
+
+		StepVerifier.create(
+						client.port(disposableServer.port())
+								.post()
+								.uri("/echo")
+								.send(ByteBufFlux.fromString(Mono.just("Hello world!")))
+								.responseContent()
+								.aggregate()
+								.asString()
+								.timeout(Duration.ofSeconds(10))
+				)
+				.expectNext("Hello world!")
+				.verifyComplete();
+
+		try {
+			// Wait till all logs are flushed
+			Thread.sleep(200);
+		} catch (InterruptedException e) {
```

Comment:
[checkstyle] catch on new row

```suggestion
		}
		catch (InterruptedException e) {
```